Gems/Atom/RPI/Code/Source/RPI.Public/Material/MaterialSystem.cpp

```diff
@@ -38,8 +42,473 @@ namespace AZ
+            }
+            if (!m_sceneMaterialSrgShaderAsset)
+            {
+                // neither the VBufferMaterial nor the fallback - asset-Ids were valid: try again, maybe the assetProcessor is still running
+                AZ_Warning("MaterialSystem", false, "Unable to locate the Material SRG shader asset, try again");
+                return false;
+            }
+
+            CreateSceneMaterialSrg();
+
+            AZ::Data::AssetBus::Handler::BusConnect(m_sceneMaterialSrgShaderAsset.GetId());
+            return true;
+        }
+
+        void MaterialSystem::CreateSceneMaterialSrg()
+        {
+            if (m_sceneMaterialSrgShaderAsset->IsReady())
+            {
+                m_sceneMaterialSrg = ShaderResourceGroup::Create(m_sceneMaterialSrgShaderAsset, AZ::Name("SceneMaterialSrg"));
+            }
+        }
+
+        //  Data::AssetBus Interface
+        void MaterialSystem::OnAssetReloaded([[maybe_unused]] AZ::Data::Asset<AZ::Data::AssetData> asset)
+        {
+            CreateSceneMaterialSrg();
+        }
+
+        void MaterialSystem::OnAssetReady([[maybe_unused]] AZ::Data::Asset<AZ::Data::AssetData> asset)
+        {
+            CreateSceneMaterialSrg();
+        }
+
+#ifndef USE_BINDLESS_SRG
+        int32_t MaterialSystem::RegisterMaterialTexture(
+            const int materialTypeIndex, const int materialInstanceIndex, const Data::Instance<Image>& image)
+        {
+            int32_t textureIndex{ -1 };
+
+            if (!image)
+            {
+                return textureIndex;
+            }
+
+            auto& materialTypeData = m_materialTypeData[materialTypeIndex];
+            auto& instanceData = materialTypeData.m_instanceData[materialInstanceIndex];
+
+            auto textureIterator = instanceData.m_materialTexturesMap.find(image->GetAssetId());
+            if (textureIterator == instanceData.m_materialTexturesMap.end())
+            {
+                textureIndex = static_cast<int32_t>(instanceData.m_materialTextures.size());
+                instanceData.m_materialTextures.emplace_back(image);
+                instanceData.m_materialTexturesDirty = true;
+            }
+            else
+            {
+                textureIndex = textureIterator->second;
+            }
+            return textureIndex;
+        }
+#endif
+
+        // MaterialSrgHandler Interface
+        MaterialInstanceData MaterialSystem::RegisterMaterialInstance(const Data::Instance<Material> material)
+        {
+            if (!m_sceneMaterialSrgShaderAsset)
+            {
+                LoadMaterialSrgShaderAsset();
+            }
+            m_bufferReadIndicesDirty = true;
+
+            int32_t materialTypeIndex{ -1 };
+
+            auto materialAsset = material->GetAsset();
+            auto materialTypeAsset = materialAsset->GetMaterialTypeAsset();
+
+            // Note: the Asset-Id of the Material-type isn't enough, i also need to check the Shader Options
```

Comment:
Where do you check the shader options?

Reply:
Nowhere, since this isn't needed yet. Updated the comment.